### PR TITLE
Derive category icon file name from category name and add unit test support.

### PR DESCRIPTION
**Pull request by jfhamlin**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -5,6 +5,10 @@ node_js:
 env:
   - CXX=g++-4.8
 
+cache:
+  directories:
+    - node_modules
```
Comment (jfhamlin) [inline .travis.yml]: Shaves a bit of time off the build.

File: .travis.yml [hunk 2]
```diff
@@ -21,6 +25,11 @@ env:
     - SANITIZED_BRANCH=$(echo $TRAVIS_BRANCH|sed 's|/|-|g')
     - REPO=sheltertechsf/askdarcel-web
 
+before_install:
+  - export CHROME_BIN=chromium-browser
+  - export DISPLAY=:99.0
+  - sh -e /etc/init.d/xvfb start
```
Comment (jfhamlin) [inline .travis.yml]: Required to allow Chrome to run for tests.

File: app/components/Category.js [hunk 3]
```diff
@@ -1,27 +1,22 @@
...
+          <img src={images.iconPath(this.props.name)} alt={this.props.name} className="img-responsive"/><br />
         </Link>
         {this.props.name}
       </div>
       <div className="hidden-lg hidden-md hidden-sm col-xs-12">
         <Link className="category_brand" to={{ pathname: "resources", query: { categoryid: this.props.categoryid } }} >
           <button className="btn btn-default category_button" type="submit">
             <div className="col-xs-4 category_xs_button_image">
-              <img src={iconPathMap[this.props.image_path]} alt={this.props.name} className="img-responsive"/><br />
+              <img src={images.iconPath(this.props.name)} alt={this.props.name} className="img-responsive"/><br />
```
Comment (jfhamlin) [inline app/components/Category.js]: Maybe @derekfidler can help here. We should be able to make this responsive in CSS without the duplication.

File: package.json [hunk 4]
```diff
@@ -35,10 +43,11 @@
     "webpack-dev-server": "^1.14.1"
   },
   "scripts": {
-    "test": "echo 'Yee-haw!'",
+    "testserver": "cross-env NODE_ENV=test karma start internals/testing/karma.conf.js",
```
Comment (jfhamlin) [inline package.json]: Use this command to keep a testing daemon running locally. Tests will be run every time a file is modified.

**Comment by jfhamlin** [comment]
@derekfidler, do you have an icon for the "Technology" category somewhere? Looks like we've been using the "work" icon till now, but a briefcase doesn't exactly scream "tech."

File: app/components/Category.js (hunk 3 above)
Comment (derekfidler) [inline app/components/Category.js]: yes i should be able to do that. removed some other duplications on other pages.

File: app/assets/index.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+});
+
+function iconPath(name) {
+  return iconPathMap[name.toLowerCase()];
+}
+
+export const images = {
+  background: require('../assets/images/bg.png'),
+  logoSmall: require('../assets/images/logo-small-white@3x.png'),
+  iconPath: iconPath
```
Comment (jfhamlin) [inline app/assets/index.js]: Kinda want to rename this to `icons`.

File: app/components/LargeHeader.js [hunk 6]
```diff
@@ -1,15 +1,14 @@
 import React from 'react';
+import { images } from '../assets';
 
-let bgImage = require('../assets/images/bg.png');
-let smallLogo = require('../assets/images/logo-small-white@3x.png');
 
 function LargeHeader() {
   return (
     <nav className="navbar navbar-default largeheader_background" style={{backgroundImage: 'url(' + bgImage +')'}}>
```
Comment (jfhamlin) [inline app/components/LargeHeader.js]: Looks like I missed replacing `bgImage`.

File: webpack.config.js [hunk 7]
```diff
@@ -47,24 +48,24 @@ module.exports = {
       {
         test: /\.(jpe?g|png|gif|svg)$/i,
         loaders: [
-          'file?hash=sha512&digest=hex&name=[hash].[ext]',
+          'file?name=[name]-[sha512:hash:hex:8].[ext]',
```
Comment (jfhamlin) [inline webpack.config.js]: This is a prettier format. An example: `ic-medical@3x-978e6ce0.png`.

**Comment by lelandg559** [comment]
so the images from the mocks dont match exactly our current categories, ill post the mapping we currently have.

**Comment by jfhamlin** [comment]
@lelandg559, rebased on master and fixed a couple more issues.

**Comment by jfhamlin** [comment]
> so the images from the mocks dont match exactly our current categories, ill post the mapping we currently have.

AFAICT the icons this change ends up using are the same as what we've got in staging except for the corrected "technology" icon.

File: webpack.config.js [hunk 8]
```diff
@@ -7,14 +7,15 @@ var ExtendedDefinePlugin = require('extended-define-webpack-plugin');
...
 
 module.exports = {
   context: __dirname,
   entry: path.resolve(appRoot, 'init.js'),
   output: {
-    path: path.resolve(__dirname, 'build'),
+    path: buildDir,
     publicPath: '/dist/',
-    filename: 'bundle.js'
+    filename: 'bundle.[hash].js'
```
Comment (lelandg559) [inline webpack.config.js]: getting

GET http://askdarcel.lvh.me:8080/dist/bundle.5d36568044b620a6bd28.js 404 (Not Found)

**Comment by lelandg559** [comment]
LGTM we can clean up api call to remove the image_path

**Pull request merged**